File: AdventurePlannerBE/ViewModels/TripDTO.cs
using AdventurePlannerBE.Models;
using System.ComponentModel.DataAnnotations;

namespace AdventurePlannerBE.ViewModels
{
    public class TripDTO
    {
        public Guid Id { get; set; }
        public string Name { get; set; }

        public DateOnly StartDate { get; set; }

        public DateOnly EndDate { get; set; }

        public int NumberOfPersons { get; set; }

        public int EstimatedBudget { get; set; }

        public TripDTO MapData(Trip trip)
        {
            Id = trip.Id;
            Name = trip.Name;
            StartDate = trip.StartDate;
            EndDate = trip.EndDate;
            NumberOfPersons = trip.NumberOfPersons;
            EstimatedBudget = trip.EstimatedBudget;

            return this;
        }
    }
}

File: AdventurePlannerFE/src/app/dtos/trip-dto.ts
import { EMPTY_GUID } from '../constants';
import { TripViewModel } from '../pages/models/trip-view-model';

export class TripDto {
  id = '';
  name = '';
  startDate = '';
  endDate = '';
  numberOfPersons = 0;
  estimatedBudget = 0;

  updateBasedOnModel(trip: TripViewModel) {
    if (this.id == '') {
      this.id = EMPTY_GUID;
    }
    
    this.name = trip.name;
    this.startDate = trip.startDate.toISOString().split('T')[0];
    this.endDate = trip.endDate.toISOString().split('T')[0];
    this.numberOfPersons = trip.numberOfPersons;
    this.estimatedBudget = trip.estimatedBudget;
  }
}
